integration/proxy_test.go

```diff
@@ -640,3 +641,71 @@ func TestALPNProxyDialProxySSHWithoutInsecureMode(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "hello world\n", output.String())
 }
+
+// TestALPNProxyHTTPProxyNOProxyDial tests if a node joining to root cluster
+// takes into account http_proxy and no_proxy env variables.
+func TestALPNProxyHTTPProxyNOProxyDial(t *testing.T) {
+	lib.SetInsecureDevMode(true)
+	defer lib.SetInsecureDevMode(false)
+
+	rc := NewInstance(InstanceConfig{
+		ClusterName: "root.example.com",
+		HostID:      uuid.New(),
+		NodeName:    Loopback,
+		log:         testlog.FailureOnly(t),
+		Ports:       singleProxyPortSetup(),
+	})
+	username := mustGetCurrentUser(t).Username
+	rc.AddUser(username, []string{username})
+
+	rcConf := service.MakeDefaultConfig()
+	rcConf.DataDir = t.TempDir()
+	rcConf.Auth.Enabled = true
+	rcConf.Auth.NetworkingConfig.SetProxyListenerMode(types.ProxyListenerMode_Multiplex)
+	rcConf.Auth.Preference.SetSecondFactor("off")
+	rcConf.Proxy.Enabled = true
+	rcConf.Proxy.DisableWebInterface = true
+	rcConf.SSH.Enabled = false
+
+	err := rc.CreateEx(t, nil, rcConf)
+	require.NoError(t, err)
+
+	err = rc.Start()
+	require.NoError(t, err)
+	defer rc.StopAll()
+
+	// Create and start http_proxy server.
+	ps := &proxyServer{}
+	ts := httptest.NewServer(ps)
+	defer ts.Close()
+
+	u, err := url.Parse(ts.URL)
+	require.NoError(t, err)
+
+	require.NoError(t, os.Setenv("http_proxy", u.Host))
+	t.Cleanup(func() { os.Unsetenv("http_proxy") })
+
+	require.NoError(t, os.Setenv("no_proxy", "127.0.0.1"))
+	t.Cleanup(func() { os.Unsetenv("no_proxy") })
```

Comment:
Use [`t.SetEnv`](https://pkg.go.dev/testing#T.Setenv)
```suggestion
	t.Setenv("http_proxy", u.Host)
	t.Setenv("no_proxy", "127.0.0.1")
```

Reply:
This is a nice trick to know.

Reply:
Thanks.